llvm/lib/Transforms/Scalar/InferAddressSpaces.cpp

```diff
@@ -1226,11 +1245,10 @@ bool InferAddressSpacesImpl::rewriteWithNewAddressSpaces(
       I = skipToNextUser(I, E);
 
       if (isSimplePointerUseValidToReplace(
-              *TTI, U, V->getType()->getPointerAddressSpace())) {
+              *TTI, CurUser, V->getType()->getPointerAddressSpace(), V, NewV)) {
         // If V is used as the pointer operand of a compatible memory operation,
```

Comment:
The comment may also be out of date/or out of place/redundant now, as all the magic happens in `isSimplePointerUseValidToReplace`.